File: InterviewHub/my-react-app/src/services/questionService.js
import axiosClient from './axiosClient';

export const questionService = {
  getAllQuestions: (params) => {
    return axiosClient.get('/questions', { params });
  },
  getQuestionById: (id) => {
    return axiosClient.get(`/questions/${id}`);
  },
  createQuestion: (data) => {
    return axiosClient.post('/questions', data);
  },
  updateQuestion: (id, data) => {
    return axiosClient.put(`/questions/${id}`, data);
  },
  deleteQuestion: (id) => {
    return axiosClient.delete(`/questions/${id}`);
  }
};

File: Backend/src/main/java/com/InterviewHub/feature/question/QuestionService.java
package com.InterviewHub.feature.question;

import com.InterviewHub.dto.PaginationDTO;
import com.InterviewHub.feature.question.dto.CreateQuestionRequest;
import com.InterviewHub.feature.question.dto.QuestionResponse;
import com.InterviewHub.feature.question.dto.UpdateQuestionRequest;

public interface QuestionService {
    PaginationDTO<QuestionResponse> searchQuestions(String title, DifficultyLevel difficulty, Long categoryId, int page, int size);
    QuestionResponse getQuestionById(Long id);
    QuestionResponse createQuestion(CreateQuestionRequest request);
    QuestionResponse updateQuestion(Long id, UpdateQuestionRequest request);
    void deleteQuestion(Long id);
}
